blocks/api/post.pegjs

```diff
@@ -148,7 +148,26 @@ WP_Block_End
   }
 
 WP_Block_Name
-  = $(ASCII_Letter (ASCII_AlphaNumeric / "/" ASCII_AlphaNumeric)*)
+  = namespace:$(WP_Block_Namespace "/")? name:WP_Namespaced_Block_Name
+  {
+    /** <?php
+    if ( ! $namespace ) {
+      $namespace = 'core/';
+    }
+    return $namespace . $name;
+    ?> **/
+
+    if ( ! namespace ) {
+      namespace = 'core/';
+    }
+    return namespace + name;
+  }
```

Comment:
We're still missing some of the value of the parser: a separation of code from grammar. The more we can eliminate processing from inside of here the more declarative it is and the easier it will be to recreate in other contexts.

```
WP_Block_Name
  = name:$(WP_Block_Namespace "/" WP_Namespaced_Block_name)
  { /** <?php return "core/$name" ?> */ return 'core/' + name }
  / $(WP_Namespaced_Block_Name)
```

Reply:
Ah, I see what you're saying. That's a much nicer way of doing it, added in 5ebaf95.